testsuite/prometheus.py

```diff
@@ -94,3 +94,21 @@ def _wait_for_scrape():
             return False
 
         assert _wait_for_scrape(), "Scrape wasn't done in time"
+
+    def verify_no_observability_targets(self, service_monitor_services: list[str], pod_monitor_jobs: list[str]) -> bool:
+        """Verify that no observability targets are active in Prometheus"""
+
+        @backoff.on_predicate(backoff.constant, interval=5, max_tries=12, jitter=None)
+        def targets_cleared():
+            targets = self.get_active_targets()
+
+            for t in targets:
+                labels = t.get("labels", {})
+                if labels.get("service") in service_monitor_services:
+                    return False
+                if labels.get("job") in pod_monitor_jobs:
+                    return False
+
+            return True
+
+        return targets_cleared()
```

Comment:
You need to generalize class methods more, their intention is to be reused. Seems like you just copied the fixture here. See if you can make this method here working only for a single servicemonitor/podmonitor (also ideally generalized (not just 2 different parameters but decision based on a value this method got)) :sweat_smile: